Wazuh 4.4.0 release notes and versions update (#5840)
wazuh/wazuh-documentation: source/release-notes/release-4-4-0.rst

.. Copyright (C) 2015, Wazuh, Inc.

.. meta::
  :description: Wazuh 4.4.0 has been released. Check out our release notes to discover the changes and additions of this release.

4.4.0 Release notes - 31 January 2023
=====================================

This section lists the changes in version 4.4.0. Every update of the Wazuh solution is cumulative and includes all enhancements and fixes from previous releases.

Highlights
----------

This new version of Wazuh brings new features and adds support related to different Linux distributions and integrations. For more details, the highlights of Wazuh 4.4.0 are listed below:

- IPv6 support for the enrollment process and the agent-manager connection
- Vulnerabilities detection support for SUSE agents

> - Expanded operating systems support with Alpine Linux agent

Reviewer: ```suggestion
- Alpine Linux agent to expand the operating systems supported
```

- Wazuh indexer and dashboard based on the latest Opensearch 2.4.1 version
- Rework of Ubuntu Linux 20.04 and 22.04 SCA policies
- Support for Azure Integration in Linux agents

Below you can find more information about each of these highlights.

Wazuh 4.4.0 brings IPv6 support when connecting and enrolling an agent to a manager. The IPv6 protocol can handle packets more effectively, enhance performance, and boost security. This new feature allows agents to register and connect through an IPV6 address.

> SUSE agents now natively support vulnerabilities detection. Wazuh added full support for SUSE Linux Enterprise Server and Desktop 11, 12, and 15 operating systems. The vulnerability Detector now scans the programs identified by ``syscollector``, looking to report vulnerabilities described in the `SUSE OVAL <https://www.suse.com/support/security/oval/>`_ and the `NVD <https://nvd.nist.gov/>`_ databases.

Reviewer: ```suggestion
SUSE agents now natively support vulnerabilities detection. Wazuh added full support for SUSE Linux Enterprise Server and Desktop operating systems versions 11, 12, and 15. The vulnerability Detector now scans the programs identified by ``syscollector``, looking to report vulnerabilities described in the `SUSE OVAL <https://www.suse.com/support/security/oval/>`_ and the `NVD <https://nvd.nist.gov/>`_ databases.
```

> Wazuh 4.4.0 adds support for Alpine Linux. As Alpine Linux is a distribution that has gained great popularity among Linux users, we saw the need for the Wazuh agent to support Alpine Linux and ensure the proper support for this distribution. We generated Alpine Linux installers to make it easier to install and thus spare users of this distribution from having to compile the source code locally.

Reviewer: ```suggestion
Wazuh 4.4.0 adds support for Alpine Linux. As Alpine Linux is a distribution that has gained great popularity among Linux users, we saw the need for the Wazuh agent to support Alpine Linux and ensure the proper support for this distribution. We generated Alpine Linux installers to make it easier to install and thus spare users from having to compile the source code locally.
```

.. thumbnail:: /images/release-notes/4.4.0/alpine-linux.png
      :align: center
      :width: 60%

>       :title: This new version adds support for Alpine Linux

Reviewer: ```suggestion
      :title: Wazuh 4.4.0 adds support for Alpine Linux
```

> Wazuh indexer and dashboard bump to OpenSearch 2.4.1. The Wazuh indexer and the Wazuh dashboard are based on OpenSearch, an open source search and analytics project derived from Elasticsearch and Kibana. We generated and tested the ``wazuh-indexer`` Debian and RPM packages with OpenSearch 2.4.1 and the ``wazuh-dashboard`` Debian and RPM packages with OpenSearch dashboards 2.4.1. This way, we avoid earlier version vulnerabilities, have all OpenSearch new functionalities, and are up to date with the OpenSearch project development.

Reviewer: ```suggestion
Wazuh indexer and dashboard based on the latest OpenSearch 2.4.1. The Wazuh indexer and the Wazuh dashboard are based on OpenSearch, an open source search and analytics project derived from Elasticsearch and Kibana. We generated and tested the ``wazuh-indexer`` Debian and RPM packages with OpenSearch 2.4.1 and the ``wazuh-dashboard`` Debian and RPM packages with OpenSearch dashboards 2.4.1. With this bump to the latest version, we avoid earlier version vulnerabilities, have all OpenSearch new functionalities, and are up to date with the OpenSearch project development.
```

To solve some errors in the previous Ubuntu Linux 20.04 SCA Policy, we reworked the Ubuntu Linux 20.04 and 22.04 SCA policies. As part of this task, we used the *CIS Ubuntu Linux 22.04 LTS Benchmark v1.0.0* to update Ubuntu Linux 22.04 SCA Policy.

> Wazuh added support for Azure Integration in Linux agents. Now this integration can run for both agents and managers. We modified the packages generation process to support Azure in those agents installed using the WPK packages. Each new WPK package contains all the updated binaries and source code, and the installer updates all files and binaries to support Azure integration.

Reviewer: ```suggestion
Wazuh added support for Azure Integration in Linux agents. Now this integration can run for both agents and managers. We modified the packages generation process to support Azure in those agents that are installed using the WPK packages. Each new WPK package contains all the updated binaries and source code, and the installer updates all files and binaries to support Azure integration.
```

.. thumbnail:: /images/release-notes/4.4.0/azure-integration.png
      :align: center
      :width: 60%

>       :title: This new version adds support Azure Integration in Linux agents

Reviewer: ```suggestion
      :title: Wazuh 4.4.0 adds support for the Azure Integration in Linux agents
```
